src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -5635,6 +5875,248 @@ interface I14
+}
+
+interface I3<T3> where T3 : I3<T3>
+{
+    static abstract " + op + @" operator T3(bool y);
+}
+
+interface I4<T4> where T4 : struct, I4<T4>
+{
+    static abstract " + op + @" operator T4?(bool y);
+}
+
+class C5<T5> where T5 : C5<T5>.I6
+{
+    public interface I6
+    {
+        static abstract " + op + @" operator T5 (bool y);
+    }
+}
+
+interface I7<T71, T72> where T72 : I7<T71, T72> where T71 : T72
+{
+    static abstract " + op + @" operator T71 (bool y);
+}
+
+interface I8<T8> where T8 : I9<T8>
+{
+    static abstract " + op + @" operator T8(bool y);
+}
+
+interface I9<T9> : I8<T9> where T9 : I9<T9> {}
+
+interface I10<T10> where T10 : C11<T10>
+{
+    static abstract " + op + @" operator T10(bool y);
+}
+
+class C11<T11> : I10<T11> where T11 : C11<T11> {}
+
+interface I12
+{
+    static abstract " + op + @" operator int(bool y);
+}
+
+interface I13
+{
+    static abstract " + op + @" operator I13(bool y);
+}
+
+interface I14<T14> where T14 : I14<T14>
+{
+    abstract static " + op + @" operator object(T14 y);
+}
+
+class C15 {}
+class C16 : C15 {}
+
+interface I17<T17> where T17 : C15, I17<T17>
+{
+    abstract static " + op + @" operator C16(T17 y);
+}
+
+interface I18<T18> where T18 : C16, I18<T18>
+{
+    abstract static " + op + @" operator C15(T18 y);
+}
+
+interface I19<T19_1, T19_2> where T19_1 : I19<T19_1, T19_2>, T19_2
+{
+    abstract static " + op + @" operator T19_1(T19_2 y);
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+            compilation1.VerifyDiagnostics(
+                // (4,39): error CS0555: User-defined operator cannot convert a type to itself
+                //     abstract static explicit operator T1(T1 y);
+                Diagnostic(ErrorCode.ERR_IdentityConversion, "T1").WithLocation(4, 39),
+                // (9,39): error CS1964: 'I2<T2>.explicit operator dynamic(T2)': user-defined conversions to or from the dynamic type are not allowed
+                //     abstract static explicit operator dynamic(T2 y);
+                Diagnostic(ErrorCode.ERR_BadDynamicConversion, "dynamic").WithArguments("I2<T2>." + op + " operator dynamic(T2)").WithLocation(9, 39),
+                // (26,43): error CS9112: User-defined conversion in an interface must convert to or from the enclosing type type parameter constrained to the enclosing type
+                //         static abstract explicit operator T5 (bool y);
+                Diagnostic(ErrorCode.ERR_AbstractConversionNotInvolvingContainedType, "T5").WithLocation(26, 43),
+                // (32,39): error CS9112: User-defined conversion in an interface must convert to or from the enclosing type type parameter constrained to the enclosing type
+                //     static abstract explicit operator T71 (bool y);
```

Comment:
I think I found the specific diagnostic behavior in this test to be confusing until I tracked down these PROTOTYPE comments:

https://github.com/AlekseyTs/roslyn/blob/5a0d9e8383af65764a22595ef34e670c0f5ec333/src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs#L637

https://github.com/AlekseyTs/roslyn/blob/5a0d9e8383af65764a22595ef34e670c0f5ec333/src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs#L639